crates/forge_domain/src/api.rs

```diff
@@ -13,7 +13,7 @@ pub trait API: Sync + Send {
 
     /// Provides information about the tools available in the current
     /// environment
-    async fn tools(&self) -> Vec<ToolDefinition>;
+    async fn tools(&self, workflow: &Workflow) -> anyhow::Result<Vec<ToolDefinition>>;
```

Comment:
workflow should be optional. If we don't provide it it should return default tools.